libr/core/cmd_zign.c

```diff
@@ -183,13 +183,26 @@ static char *getFcnComments(RCore *core, RAnalFunction *fcn) {
 #endif
 
 static void addFcnZign(RCore *core, RAnalFunction *fcn, const char *name) {
+	char *zignspace = NULL;
 	char *zigname = NULL;
 	const RSpace *curspace = r_spaces_current (&core->anal->zign_spaces);
+	int i, len = 0;
 
 	if (name) {
 		zigname = r_str_new (name);
 	} else {
-		if (curspace) {
+		// If the user has set funtion names containing a single ':' then we assume
+		// ZIGNSPACE:FUNCTION, and for now we only support the 'zg' command
+		if (r_str_char_count (fcn->name, ':') == 1) {
+			for (i = 0; i < strlen (fcn->name); i++) {
```

Comment:
uhm. what about just using strchr ?

Reply:
Yeah makes much more sense my C is incredibly rust atm, worth checking I have not messed up the arithmetic there.